frontend/src/pages/hardwareProfiles/HardwareProfileEnableToggle.tsx

```diff
@@ -3,18 +3,21 @@ import { Switch } from '@patternfly/react-core';
 import useNotification from '~/utilities/useNotification';
 import { toggleHardwareProfileEnablement } from '~/api';
 import { HardwareProfileKind } from '~/k8sTypes';
+import { HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE } from './nodeResource/const';
+import { hardwareProfileWarning } from './utils';
 
 type HardwareProfileEnableToggleProps = {
   hardwareProfile: HardwareProfileKind;
-  refreshHardwareProfiles: () => void;
 };
 
 const HardwareProfileEnableToggle: React.FC<HardwareProfileEnableToggleProps> = ({
   hardwareProfile,
-  refreshHardwareProfiles,
 }) => {
-  const { enabled } = hardwareProfile.spec;
-  const [isEnabled, setEnabled] = React.useState(enabled);
+  const hardwareProfileWarningMessage = hardwareProfileWarning(hardwareProfile);
+  const enabled =
+    hardwareProfile.spec.enabled &&
+    (typeof hardwareProfileWarningMessage === 'undefined' ||
+      hardwareProfileWarningMessage.message.includes(HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE));
```

Comment:
i would keep is enabled separate from the warning. Just use warning to additionally disable the hardware profile